test/config-luatest/upgrade_test.lua

```diff
@@ -71,14 +71,13 @@ end)
 g.test_upgrade = function()
     g.server:exec(function()
         t.assert_equals(box.space._schema:get("version"), {'version', 2, 11, 0})
-        local messages = {}
-        for _, alert in pairs(require("config"):info().alerts) do
-            table.insert(messages, alert.message)
-        end
-        local exp = 'box.schema.user.grant("guest", "read", "space", ' ..
-            '"_space") has failed because either the object has not been ' ..
-            'created yet, a database schema upgrade has not been performed, ' ..
-            'or the privilege write has failed (separate alert reported)'
-        t.assert_items_include(messages, {exp})
+
+        -- We will change 'guest' rights, so need to change user
+        box.session.su('admin')
+        box.schema.upgrade()
+        -- After upgrade privileges should be granted automatically
+
+        local exp2 = { 'read', 'space', '_space' }
+        t.assert_items_include(box.schema.user.info('guest'), {exp2})
```

Comment:
I would like if we check the following scenarios too:

* A user/a role is added.
* A password of a user is changed.

Reply:
Done all of the suggested! Thanks for the note.